engine/robot_state.py

```diff
@@ -0,0 +1,138 @@
+            lb_ultrasonic: the ultrasonic at the back of the left side of the robot, used for boundary following
+            rf_ultrasonic: the ultrasonic at the front of the right side of the robot, used for boundary following
+            rb_ultrasonic: the ultrasonic at the back of the right side of the robot, used for boundary following
+            is_sim: False if the physical robot is being used, True otherwise
+            is_store: False if csv data should not be stored, True otherwise
+            position_kp: the proportional factor of the position PID
+            position_ki: the integral factor of the position PID
+            position_kd: the derivative factor of the position PID
+            position_noise: the flat amount of noise added to the robot's phase on each localization step
+            heading_kp: the proportional factor of the heading PID
+            heading_ki: the integral factor of the heading PID
+            heading_kd: the derivative factor of the heading PID
+            heading_noise: ?
+            phase: the phase of the robot
+            time_step: the amount of time that passes between each feedback loop cycle, should only be used if is_sim
+                is True
+            move_dist: the distance in meters that the robot moves per time dt
+            turn_angle: the angle in radians that the robot turns per time dt regardless of time step
+            plastic_weight: the weight of the trash the robot has collected
+            init_threshold (Double): Radius from initial position that will detect robot is back in initial position
+            goal_threshold (Double): Threshold from goal that will be detected as reaching goal in obstacle avoidance
+            noise_margin (Double): Margin from init_threshold that the robot has to leave before detecting robot has
+                left initial position
+        """
+        # User defined parameters
+        self.epsilon = epsilon
+        self.max_velocity = max_velocity
+        self.radius = radius
+
+        # Flags
+        self.is_sim = not is_raspberrypi()
+        self.is_store = False
+        self.phase = Phase.SETUP
+        self.avoid_obstacle = False
+        
+        # Constants
+        self.width = 700
+        self.time_step = 1
+        self.position_kp = 1
+        self.position_ki = 0
+        self.position_kd = 0
+        self.position_noise = 0
+        self.heading_kp = 1
+        self.heading_ki = 0
+        self.heading_kd = 0
+        self.heading_noise = 0
+        self.move_dist = 0.5
+        self.turn_angle = 3
+        self.turn_angle = self.turn_angle / self.time_step # dividing by time_step ignores the effect of time_step on absolute
+        self.init_threshold = 1
+        self.goal_threshold = 1
+        self.noise_margin = 1
+        self.front_sensor_offset = 0  # TODO: replace this with how far offset the sensor is to the front of the robot
+        self.max_sensor_range = 600
```

Comment:
this is not in the specification above - what unit is this?

Reply:
front_sensor_offset and max_sensor_range in cm, can prob be moved into the obstacle avoidance thread instead of being a class attribute bc only rly being used in that function/thread. idk, which makes more sense?